lightning/src/blinded_path/message.rs

```diff
@@ -148,7 +148,14 @@ pub(super) fn blinded_hops<T: secp256k1::Signing + secp256k1::Verification>(
 		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { next_hop, next_blinding_override: None }))
 		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })));
 
-	utils::construct_blinded_hops(secp_ctx, pks, tlvs, session_priv)
+	let max_length = tlvs.clone()
+		.max_by_key(|c| c.serialized_length())
+		.map(|c| c.serialized_length())
+		.unwrap_or(0);
+
+	let length_tlvs = tlvs.map(|tlv| (max_length, tlv));
```

Comment:
Let's make a wrapper struct instead of using a tuple so that this is more readable.

Reply:
Makes sense!
Introduce the wrapper structs in [pr3177.04](https://github.com/shaavan/rust-lightning/commits/pr3177.04)
Thanks!